### PR TITLE
python3.pkgs.heatshrink2: init at 0.11.0

### PR DESCRIPTION
**Pull request by adisbladis**
###### Description of changes
This is used in the firmware build for Flipper Zero which I'm attempting to package.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/heatshrink2/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{ buildPythonPackage, fetchPypi, lib }:
+
+let
+  pname = "heatshrink2";
+  version = "0.11.0";
+
+in
+buildPythonPackage {
+  inherit pname version;
```
Comment (pbsds) [inline pkgs/development/python-modules/heatshrink2/default.nix:9]:
```suggestion
buildPythonPackage rec {
  pname = "heatshrink2";
  version = "0.11.0";
```

**Review by pbsds — CHANGES_REQUESTED** [review]

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 182185` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.heatshrink2</li>
    <li>python39Packages.heatshrink2</li>
  </ul>
</details>

The package runs its tests properly. With the suggested change it LGTM!

<details>
<summary>resulting paths:</summary>

```
.../lib
.../lib/python3.10
.../lib/python3.10/site-packages
.../lib/python3.10/site-packages/heatshrink2
.../lib/python3.10/site-packages/heatshrink2/__init__.py
.../lib/python3.10/site-packages/heatshrink2/__main__.py
.../lib/python3.10/site-packages/heatshrink2/__pycache__
.../lib/python3.10/site-packages/heatshrink2/__pycache__/__init__.cpython-310.pyc
.../lib/python3.10/site-packages/heatshrink2/__pycache__/__main__.cpython-310.pyc
.../lib/python3.10/site-packages/heatshrink2/__pycache__/streams.cpython-310.pyc
.../lib/python3.10/site-packages/heatshrink2/__pycache__/version.cpython-310.pyc
.../lib/python3.10/site-packages/heatshrink2/core.cpython-310-x86_64-linux-gnu.so
.../lib/python3.10/site-packages/heatshrink2/streams.py
.../lib/python3.10/site-packages/heatshrink2/version.py
.../lib/python3.10/site-packages/heatshrink2-0.11.0.dist-info
.../lib/python3.10/site-packages/heatshrink2-0.11.0.dist-info/INSTALLER
.../lib/python3.10/site-packages/heatshrink2-0.11.0.dist-info/LICENSE
.../lib/python3.10/site-packages/heatshrink2-0.11.0.dist-info/METADATA
.../lib/python3.10/site-packages/heatshrink2-0.11.0.dist-info/RECORD
.../lib/python3.10/site-packages/heatshrink2-0.11.0.dist-info/REQUESTED
.../lib/python3.10/site-packages/heatshrink2-0.11.0.dist-info/WHEEL
.../lib/python3.10/site-packages/heatshrink2-0.11.0.dist-info/direct_url.json
.../lib/python3.10/site-packages/heatshrink2-0.11.0.dist-info/top_level.txt
.../nix-support
.../nix-support/propagated-build-inputs
```
</details>

File: pkgs/development/python-modules/heatshrink2/default.nix (hunk 1 above)
Comment (adisbladis) [reply pkgs/development/python-modules/heatshrink2/default.nix:9]: I prefer creating a `let` block over using recursive attrsets.

File: pkgs/development/python-modules/heatshrink2/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-xihtHUuS+664Ow/YtfgxyTUNzRBG53blqqg9G+Q4Nfc=";
+  };
+
+  meta = {
+    description = "Compression using the Heatshrink algorithm in Python 3.";
+    homepage = "https://github.com/eerimoq/pyheatshrink";
+    license = lib.licenses.isc;
+    maintainers = [ ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/heatshrink2/default.nix:20]: This must be set

File: pkgs/development/python-modules/heatshrink2/default.nix [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+buildPythonPackage {
+  inherit pname version;
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-xihtHUuS+664Ow/YtfgxyTUNzRBG53blqqg9G+Q4Nfc=";
+  };
+
+  meta = {
+    description = "Compression using the Heatshrink algorithm in Python 3.";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/heatshrink2/default.nix:17]:
```suggestion
    description = "Compression using the Heatshrink algorithm";
```

File: pkgs/development/python-modules/heatshrink2/default.nix [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+in
+buildPythonPackage {
+  inherit pname version;
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-xihtHUuS+664Ow/YtfgxyTUNzRBG53blqqg9G+Q4Nfc=";
+  };
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/heatshrink2/default.nix:15]: There are tests https://github.com/eerimoq/pyheatshrink/tree/master/tests . Are those being executed? Regardless of that we should add a pythonImportsCheck.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**